phoenix-core/src/main/java/org/apache/phoenix/compile/FromCompiler.java

```diff
@@ -163,16 +163,22 @@ public static ColumnResolver getResolverForCreation(final CreateTableStatement s
     	
         TableName baseTable = statement.getBaseTableName();
         String schemaName;
-        if (baseTable == null) {
-            if (SchemaUtil.isSchemaCheckRequired(statement.getTableType(), connection.getQueryServices().getProps())) {
-                schemaName = statement.getTableName().getSchemaName();
-                if (schemaName != null) {
-                    new SchemaResolver(connection, statement.getTableName().getSchemaName(), true);
-                } else if (connection.getSchema() != null) {
-                    // To ensure schema set through properties or connection string exists before creating table
-                    new SchemaResolver(connection, connection.getSchema(), true);
-                }
+        if (SchemaUtil.isSchemaCheckRequired(statement.getTableType(),
+                connection.getQueryServices().getProps())) {
+            schemaName = statement.getTableName().getSchemaName();
+            if (schemaName != null) {
+                new SchemaResolver(connection,
```

Comment:
This is not an intuitive way of checking for the existence of a schema.
Please add a comment with some explanation.